src/main/scala/com/monsanto/arch/cloudformation/model/resource/ElasticLoadBalancing.scala

```diff
@@ -4,43 +4,254 @@ import com.monsanto.arch.cloudformation.model._
 import spray.json._
 
 /**
- * Created by Ryan Richt on 2/28/15
+ * The AvailabilityZones and Subnets properties are mutually exclusive, with the latter
+ * being used for ELBs within VPCs.  Therefore, the default constructor is private and
+ * methods on the companion object are used to safely construct the resource.  See
+ * http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-elb.html
  */
 
-case class `AWS::ElasticLoadBalancing::LoadBalancer`(
-  name:           String,
-  CrossZone:      Boolean,
-  SecurityGroups: Seq[Token[ResourceRef[`AWS::EC2::SecurityGroup`]]],
-  Subnets:        Seq[Token[ResourceRef[`AWS::EC2::Subnet`]]],
-  Listeners:      Seq[Listener],
-  HealthCheck:    HealthCheck,
-  Policies:       Option[Seq[LoadBalancerPolicy]],
-  Tags:           Seq[AmazonTag],
-  Scheme:         Option[String] = None, // TODO: Make this an enum
-  override val Condition: Option[ConditionRef] = None
-  ) extends Resource[`AWS::ElasticLoadBalancing::LoadBalancer`]{
-
+case class `AWS::ElasticLoadBalancing::LoadBalancer` private (
+  name:                      String,
+  Listeners:                 Seq[ELBListener],
+  AccessLoggingPolicy:       Option[ELBAccessLoggingPolicy],
+  AppCookieStickinessPolicy: Option[ELBAppCookieStickinessPolicy],
+  AvailabilityZones:         Option[Token[String]],
+  ConnectionDrainingPolicy:  Option[ELBConnectionDrainingPolicy],
+  ConnectionSettings:        Option[ELBConnectionSettings],
+  CrossZone:                 Option[Boolean],
+  HealthCheck:               Option[ELBHealthCheck],
+  Instances:                 Option[Seq[Token[ResourceRef[`AWS::EC2::Instance`]]]],
```

Comment:
Option[Seq[T]] looks weird to me. I understand this is to avoid deserialization problems due to the Amazon API though. The other option is to write custom deserializers which is a bit cumbersome. Anyone know of a better solution?